adafruit_minimqtt/adafruit_minimqtt.py

```diff
@@ -398,18 +431,15 @@ def ping(self):
         there is an active network connection.
         """
         self.is_connected()
+        if self._pending_ping_response:
+            self._pending_ping_response = False
```

Comment:
Instead of handling the polling for `PINGRESP` within `wait_for_msg`, we should be handling it within this method. 

Otherwise, any method calling `ping`, such as user-code within `code.py` will need to implement code to poll for a pingresp.

> Also, ping method tries to drain buffer looking for the PINGRESP message and if that is not the message in the
receive buffer, it never reverts the 1 byte it took out of the buffer. via https://github.com/adafruit/Adafruit_CircuitPython_MiniMQTT/issues/54

Could we implement this within `ping()` instead?

Reply:
So that is the thing. Imagine that just before we send this ping a message that is not "pingresp" arrives. We would be duplicating wait_for_msg in order to fish the pingresp out of the list of unrelated messages. Having a flag to indicate whether the ping arrived is a cleaner way of checking for the response, don't you thinkk?

Reply:
One thing we could do is to add something like:

```
other_msgs = []
while not self._pending_ping_response:
    msg = self.wait_for_msg()
    if msg:
        other_msgs.append(msg)
    if <time_to_give_up>:
        raise <ping response did not make it...>

return other_msgs
```

Would that be better?